README.md
# Aircraft-Combat
Game developing practice based on Pygame


CHANGELOG.md
**2022-06-08**  
   1.Added shop  
   2.Added score counter system  
   
**2022-07-19**  
   1.Improved object construstion  
   2.Added resource file which store the necessary data/resource  


object.py
import pygame
import os
import random
from resource import *

# object class


class Player(pygame.sprite.Sprite):
    # sprite for the Player
    # default construction: img=player_img, hp=10, speed=2, load=1000, fuel=5000, speed_limit=5
    def __init__(self, img=player_img, hp=10, speed=2, load=1000, fuel=5000, speed_limit=5):
        # this line is required to properly create the sprite
        pygame.sprite.Sprite.__init__(self)

        # create a plain rectangle for the sprite image
        self.image = img
        #self.image.set_colorkey(BLACK)

        # find the rectangle that encloses the image
        self.rect = self.image.get_rect()

        # center the sprite on the screen
        self.rect.center = (self.rect.width / 2, HEIGHT / 2)

        # player's speed
        self.default_speed = speed
        self.speed_limit = speed_limit
        self.speed = speed

        # amo
        self.gun_pos = None

        self.bullets = pygame.sprite.Group()
        self.load = load
        self.load_full = load

        # hit point
        self.hp = hp
        self.hp_full = hp
        self.shield = 10

        self.lifebar = Lifebar(self)

        # fuel
        self.fuel = fuel
        self.fuel_full = fuel
        self.speed_up = False

    def reset(self):
        self.__init__()

    def draw(self, surface):
        surface.blit(self.image, (self.rect.x, self.rect.y))
        # self.lifebar.paint(surface)

    def update(self):
        # any code here will happen every time the game loop updates
        # self.rect.x += 1

        gun_x = self.rect.x + 138
        gun_y = self.rect.y + 62
        self.gun_pos = (gun_x, gun_y)

        if self.bullets is not None:
            for bullet in self.bullets:
                bullet.update()

        self.hp = min(self.hp, self.hp_full)
        self.load = min(self.load, self.load_full)
        self.fuel = min(self.fuel, self.fuel_full)

        if self.fuel <= 0:
            self.speed_up = False
        if self.speed_up:
            self.fuel -= 1
            self.speed = self.speed_limit
        else:
            self.speed = self.default_speed

    def jump(self):
        self.speed_up = False if self.speed_up else True

    def move(self, direction):
        if direction == "up":
            self.rect.y -= self.speed
        if direction == "down":
            self.rect.y += self.speed
        if direction == "left":
            self.rect.x -= self.speed
        if direction == "right":
            self.rect.x += self.speed

        if self.rect.x > WIDTH - 150:
            self.rect.x = WIDTH - 150
        if self.rect.x < -50:
            self.rect.x = -50
        if self.rect.y < 0:
            self.rect.y = 0
        if self.rect.y > HEIGHT:
            self.rect.y = HEIGHT


class Enemy(pygame.sprite.Sprite):
    # sprite for the Player
    # default construction: image=enemy_img, speed=2, hp=1
    def __init__(self, image=enemy_img, speed=2, hp=1):
        # this line is required to properly create the sprite
        pygame.sprite.Sprite.__init__(self)

        # create a plain rectangle for the sprite image
        self.image = image
        #self.image.set_colorkey(BLACK)

        # find the rectangle that encloses the image
        self.rect = self.image.get_rect()
        # self.mask = pygame.mask.from_surface(self.image)

        # center the sprite on the screen
        x = random.randint(int(WIDTH/2), WIDTH)
        y = random.randint(100, HEIGHT)
        self.rect.center = (x, y)

        # enemy's speed
        self.speed = speed

        # amo
        x = self.rect.x + 15
        y = self.rect.y + 95
        self.gun_pos = (x, y)

        self.bullets = pygame.sprite.Group()

        # HP
        self.hp = hp
        self.hp_full = hp
        self.lifebar = Lifebar(self)

    def update(self):
        x = self.rect.x + 15
        y = self.rect.y + 95
        self.gun_pos = (x, y)

        self.ai_movement()
        if self.bullets is not None:
            for bullet in self.bullets:
                bullet.update()

    def ai_movement(self):
        self.rect.x -= 1
        x = random.randint(0, 1000)
        if x < 10:
            bullet = Bullet(self, -1)
            self.bullets.add(bullet)

    def draw(self, surface):
        surface.blit(self.image, (self.rect.x, self.rect.y))

        self.lifebar.paint(surface)


class Boss(pygame.sprite.Sprite):
    # sprite for the Player
    def __init__(self, image=boss_img, speed=2, direction=1, hp=20):
        # this line is required to properly create the sprite
        pygame.sprite.Sprite.__init__(self)

        # create a plain rectangle for the sprite image
        self.image = image
        #self.image.set_colorkey(BLACK)

        # find the rectangle that encloses the image
        self.rect = self.image.get_rect()
        # self.mask = pygame.mask.from_surface(self.image)

        # center the sprite on the screen

        self.rect.center = (1600, 500)

        # enemy's spped
        self.speed = speed
        self.direction = direction

        # amo
        self.gun_pos = None

        self.bullets = pygame.sprite.Group()

        # hp
        self.hp = hp
        self.hp_full = hp

        self.lifebar = Lifebar(self)

    def update(self):
        gun_x = self.rect.x + 15
        gun_y = self.rect.y + 235
        self.gun_pos = (gun_x, gun_y)

        self.ai_movement()
        if self.bullets is not None:
            for bullet in self.bullets:
                bullet.update()

    def ai_movement(self):

        x = random.randint(0, 1000)

        if x < 20:
            self.direction *= -1

        if 0 <= self.rect.y <= HEIGHT:
            self.rect.y += self.direction * self.speed
        else:
            if self.rect.y < 0:
                self.rect.y = HEIGHT
            else:
                self.rect.y = 0

        if x < 20:
            bullet = Bullet(self, -1)
            self.bullets.add(bullet)

    def draw(self, surface):
        surface.blit(self.image, (self.rect.x, self.rect.y))
        self.lifebar.paint(surface)


class Bullet(pygame.sprite.Sprite):
    # sprite for the Player
    def __init__(self, owner, direction=1):
        # this line is required to properly create the sprite
        pygame.sprite.Sprite.__init__(self)

        # create a plain rectangle for the sprite image
        if direction == 1:
            self.image = bullet_img
        else:
            self.image = enemy_bullet_img
        #self.image.set_colorkey(BLACK)

        # find the rectangle that encloses the image
        self.rect = self.image.get_rect()
        # self.mask = pygame.mask.from_surface(self.image)

        # center the sprite on the screen
        self.owner = owner

        self.rect.center = self.owner.gun_pos

        # movement
        self.speed = 3
        self.direction = direction

    def draw(self, surface):
        surface.blit(self.image, (self.rect.x, self.rect.y))

    def update(self):
        self.rect.x += self.speed * self.direction
        if self.rect.x > WIDTH:
            self.kill()


class Lifebar(pygame.sprite.Sprite):

    def __init__(self, owner):
        pygame.sprite.Sprite.__init__(self)

        self.owner = owner

    def paint(self, surface):

        if self.owner.hp_full < 3:
            return

        percent = self.owner.hp / self.owner.hp_full * 1.0
        pygame.draw.rect(surface, (0, 0, 0), (self.owner.rect.x-20, self.owner.rect.y-30, self.owner.rect.width - 2, 5))
        if percent < 0.3:
            color = RED
        elif percent > 0.8:
            color = GREEN
        else:
            color = YELLOW
        pygame.draw.rect(surface, color,
                         (self.owner.rect.x-20, self.owner.rect.y-30, int(self.owner.rect.width * percent), 5))


class Prop(pygame.sprite.Sprite):
    def __init__(self,):
        # this line is required to properly create the sprite
        pygame.sprite.Sprite.__init__(self)

        # create a plain rectangle for the sprite image
        event_id = random.randint(1, 3)
        self.event_id = event_id
        try:
            self.image = Event_img[event_id]
        except KeyError:
            raise KeyError
        self.image.set_colorkey(BLACK)

        # find the rectangle that encloses the image
        self.rect = self.image.get_rect()
        # self.mask = pygame.mask.from_surface(self.image)

        # center the sprite on the screend
        x = random.randint(0, WIDTH)
        y = random.randint(100, HEIGHT)
        self.rect.center = (x, y)

        # event_id

    def draw(self, surface):
        surface.blit(self.image, (self.rect.x, self.rect.y))

resource.py
import pygame
import os

WIDTH = 1920  # 游戏窗口的宽度
HEIGHT = 1200  # 游戏窗口的高度
FPS = 1000  # 帧率

# Colors (R, G, B)
BLACK = (0, 0, 0)
WHITE = (255, 255, 255)
RED = (255, 0, 0)
GREEN = (0, 255, 0)
BLUE = (0, 0, 255)
YELLOW = (255, 215, 0)
GREY = (28, 28, 28)

# player & enemy image resource

game_folder = os.path.dirname(__file__)

img_folder = os.path.join(game_folder, 'img')

player_img = pygame.image.load(os.path.join(img_folder, 'player1.png'))
player_img = pygame.transform.rotozoom(player_img, 0, 0.1)

enemy_img = pygame.image.load(os.path.join(img_folder, 'enemy.png'))
enemy_img = pygame.transform.rotozoom(enemy_img, 135, 0.1)

bullet_img = pygame.image.load(os.path.join(img_folder, 'bullet2.png'))
bullet_img = pygame.transform.rotozoom(bullet_img, 0, 0.08)

enemy_bullet_img = pygame.image.load(os.path.join(img_folder, 'bullet1.png'))
enemy_bullet_img = pygame.transform.rotozoom(enemy_bullet_img, 0, 0.2)

boss_img = pygame.image.load(os.path.join(img_folder, 'boss.png'))
boss_img = pygame.transform.rotozoom(boss_img, 0, 1)


# prop image resource
amo_img = pygame.image.load(os.path.join(img_folder, 'amo.png'))
amo_img = pygame.transform.rotozoom(amo_img, 0, 0.1)

hp_img = pygame.image.load(os.path.join(img_folder, 'hp.png'))
hp_img = pygame.transform.rotozoom(hp_img, 0, 0.15)

fuel_img = pygame.image.load(os.path.join(img_folder, 'fuel.png'))
fuel_img = pygame.transform.rotozoom(fuel_img, 0, 0.15)

Event_img = dict()
Event_img[1] = hp_img
Event_img[2] = amo_img
Event_img[3] = fuel_img

# video resource

video_folder = os.path.join(game_folder, 'video')
first_video = os.path.join(video_folder, 'Apex Legends.mp4')

# resource
game_folder = os.path.dirname(__file__)
img_folder = os.path.join(game_folder, 'img')
bg_img = pygame.image.load(os.path.join(img_folder, 'bg.jpg'))
bg_img = pygame.transform.scale(bg_img, (1920, 1200))

hpBar_img = pygame.image.load(os.path.join(img_folder, 'bar1.png'))
hpBar_img = pygame.transform.rotozoom(hpBar_img, 0, 0.8)

music_folder = os.path.join(game_folder, 'music')

# game setting
winPoint = 25
NormalEnemyCreatePossibility = 3
NormalPropCreatePossibility = 1
prop_hp = 2
prop_load = 20
prop_fuel = 500

main.py
import pygame
import random
import time
import sys

from pygame import KEYDOWN, K_SPACE, K_a, KEYUP, QUIT, K_ESCAPE

from resource import *
from object import *

count = 0
BOSS = False
switch = True
boss_gap = 3000
start = time.time()
score = 0

# initialize game
pygame.init()
pygame.mixer.init()  # 声音初始化

screen = pygame.display.set_mode((WIDTH, HEIGHT))
pygame.display.set_caption("1")
clock = pygame.time.Clock()

# initialize player and other groups
player = Player()
enemies = pygame.sprite.Group()
all_boss = pygame.sprite.Group()
props = pygame.sprite.Group()


def main():

    global player, BOSS, count, score, switch
    BOSS = False
    pygame.event.clear()
    player.reset()
    # initialize pygame and create window
    # show gamestart screen
    game_start()

    count = 0  # reset time counter
    switch = True  # turn the bgm switch trigger to True

    #  run the game
    run()

    # quit
    terminate()

    return


#  check game quit
def check_quit():
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            pygame.event.clear()
            terminate()
    return


# check speed up
def check_jump():
    pressed = pygame.key.get_pressed()
    if pressed[pygame.K_SPACE]:
        pygame.event.clear()
        return True


# check fire(left mouse button click or key q)
def check_shot():

    state = pygame.mouse.get_pressed()
    pressed = pygame.key.get_pressed()
    res = state[0] or pressed[pygame.K_q]
    pygame.event.clear()
    return res


# check player moving
def check_move():
    pressed = pygame.key.get_pressed()
    res = None
    if pressed[pygame.K_w]:
        res = "up"
    elif pressed[pygame.K_a]:
        res = "left"
    elif pressed[pygame.K_s]:
        res = "down"
    elif pressed[pygame.K_d]:
        res = "right"

    pygame.event.clear()
    return res


# check win
def check_win():
    if score > winPoint:
        return True


# check mouse click and return the clicked position
def check_click():
    for event in pygame.event.get():
        if event.type == pygame.MOUSEBUTTONUP:
            res = event.pos
            pygame.event.clear()
            return res

    return None


# check Esc click
def check_esc():
    pressed = pygame.key.get_pressed()
    if pressed[pygame.K_ESCAPE]:
        pygame.event.clear()
        return True


# check Space press
def checkForKeyPress():

    keyUpEvents = pygame.event.get(KEYUP)
    if len(keyUpEvents) == 0:
        return None
    if keyUpEvents[0].key == K_SPACE:
        return True


# check collision between objects
def check_collision():

    global BOSS, score, switch
    #  get all bullets from enemies (include boss)
    all_bullets = getEnemyBullets()

    #  check for the collision between player's bullet and enemies'(include boss)
    for bullet in player.bullets:
        for enemy in enemies:
            if pygame.sprite.collide_rect_ratio(0.4)(bullet, enemy):
                bullet.kill()
                enemy.hp -= 1
                if enemy.hp <= 0:
                    enemy.kill()
                    score += 1

        for boss in all_boss:
            if pygame.sprite.collide_rect_ratio(0.4)(bullet, boss):
                bullet.kill()
                boss.hp -= 1
                if boss.hp <= 0:
                    boss.kill()
                    score += 10

                    BOSS = False
                    switch = True

    #  check for the collision between enemies' bullet and player
    for enemy in enemies:
        if pygame.sprite.collide_rect_ratio(0.2)(player, enemy):
            enemy.kill()
            score += 1

            player.hp -= 1
        for bullet in enemy.bullets:
            if pygame.sprite.collide_rect_ratio(0.2)(player, bullet):
                bullet.kill()
                player.hp -= 1

    for enemy in all_boss:
        if pygame.sprite.collide_rect_ratio(0.2)(player, enemy):
            enemy.hp -= 1
            player.hp -= 1
            if enemy.hp <= 0:
                enemy.kill()
                score += 10

                BOSS = False
                switch = True

        for bullet in enemy.bullets:
            if pygame.sprite.collide_rect_ratio(0.2)(player, bullet):
                bullet.kill()
                player.hp -= 1

    #  check for the collision between bullets
    for bullet in player.bullets:
        for enemy_b in all_bullets:
            if pygame.sprite.collide_rect_ratio(1)(bullet, enemy_b):
                bullet.kill()
                enemy_b.kill()

    # check for the collision between player and prop
    for prop in props:
        if pygame.sprite.collide_rect_ratio(1)(player, prop):
            prop_event(player, prop.event_id)
            prop.kill()

    return


# create enemy
def create_enemy():
    global BOSS, all_boss

    x = random.randint(0, 1000)

    if x < NormalEnemyCreatePossibility and not BOSS:
        enemy = Enemy()
        enemies.add(enemy)

    if count % boss_gap == 0 and not BOSS and count != 0:
        boss = Boss()
        BOSS = True
        all_boss.add(boss)

        pygame.mixer.music.load(os.path.join(music_folder, 'boss_music.mp3'))
        pygame.mixer.music.play()
        pygame.mixer.music.set_volume(0.01)


# create prop
def create_prop():
    x = random.randint(0, 1000)
    if x < NormalPropCreatePossibility:
        prop = Prop()
        props.add(prop)


# process prop event
def prop_event(target, event_id):

    # process event
    if event_id == 1:
        target.hp += prop_hp
    elif event_id == 2:
        target.load += prop_load
    elif event_id == 3:
        target.fuel += prop_fuel

    player.update()


# get all bullets from enemies
def getEnemyBullets():
    all_bullets = pygame.sprite.Group()

    for enemy in enemies:
        for bullet in enemy.bullets:
            all_bullets.add(bullet)

    for boss in all_boss:
        for bullet in boss.bullets:
            all_bullets.add(bullet)

    return all_bullets


# get all bullets(enemies and player)
def getAllBullets():
    all_bullets = pygame.sprite.Group()

    for enemy in enemies:
        for bullet in enemy.bullets:
            all_bullets.add(bullet)

    for boss in all_boss:
        for bullet in boss.bullets:
            all_bullets.add(bullet)

    for bullet in player.bullets:
        all_bullets.add(bullet)

    return all_bullets


# game start screen
def game_start():

    # play start bgm
    pygame.mixer.music.load(os.path.join(music_folder, 'start_music.mp3'))
    pygame.mixer.music.play()
    pygame.mixer.music.set_volume(0.2)

    '''screen.blit(bg_img, (0, 0))

    myFont = pygame.font.Font(None, 80)
    loadImage = myFont.render("Press SPACE to Start", True, WHITE)
    screen.blit(loadImage, (650, 1000))

    pygame.display.update()
    pygame.time.wait(1000)

    pygame.event.clear()

    while True:
        if checkForKeyPress():
            break'''

    screen.blit(bg_img, (0, 0))

    start_x = 200
    start_y = 400
    gap = 100
    length = 200

    startFont = pygame.font.Font(None, 80)

    startImage = startFont.render("Start", True, WHITE)
    screen.blit(startImage, (start_x, start_y + 0 * gap))

    startImage = startFont.render("Shop", True, WHITE)
    screen.blit(startImage, (start_x, start_y + 1 * gap))

    startImage = startFont.render("Setting", True, WHITE)
    screen.blit(startImage, (start_x, start_y + 2 * gap))

    startImage = startFont.render("Exit", True, WHITE)
    screen.blit(startImage, (start_x, start_y + 3 * gap))

    pygame.display.update()
    pygame.event.clear()

    while True:
        check_quit()

        loc = check_click()
        if loc is not None:
            if loc[0] in range(start_x, start_x + length) and loc[1] in range(400, 470):  # start
                return
            if loc[0] in range(start_x, start_x + length) and loc[1] in range(480, 550):  # shop
                show_shop()
            if loc[0] in range(start_x, start_x + length) and loc[1] in range(560, 620):  # setting
                player_setting()
                game_start()
            if loc[0] in range(start_x, start_x + length) and loc[1] in range(640, 720):  # exit
                terminate()


# shop
def show_shop():
    screen.blit(bg_img, (0, 0))

    start_x = 200
    start_y = 400
    gap = 100
    length = 200

    startFont = pygame.font.Font(None, 80)

    startImage = startFont.render("EMPTY NOW", True, WHITE)
    screen.blit(startImage, (800, 500))

    total_score = get_score()

    hpFont = pygame.font.Font(None, 80)
    hpImage = hpFont.render("   Score: " + str(int(total_score)), True, WHITE)
    screen.blit(hpImage, (100, 100))

    pygame.display.update()
    pygame.event.clear()

    while True:
        if check_quit():
            terminate()

        if check_esc():
            main()

        loc = check_click()
        if loc is not None:
            pass


# player setting (level up)
def player_setting():

    screen.blit(bg_img, (0, 0))

    myFont = pygame.font.Font(None, 80)

    start_x = 200
    start_y = 400
    gap = 100
    length = 200

    loadImage = myFont.render("Load", True, WHITE)
    screen.blit(loadImage, (start_x, start_y + 0 * gap))

    loadImage = myFont.render("Back", True, WHITE)
    screen.blit(loadImage, (start_x, start_y + 1 * gap))

    pygame.display.update()
    pygame.event.clear()

    while True:
        if check_quit():
            terminate()

        if check_esc():
            main()

        loc = check_click()
        if loc is not None:
            if loc[0] in range(start_x, start_x + length) and loc[1] in range(400, 500):  # loading player
                load()
                return
            if loc[0] in range(start_x, start_x + length) and loc[1] in range(500, 600):  # back
                return


# game over
def game_over():
    check_quit()

    # play game over music
    pygame.mixer.music.load(os.path.join(music_folder, 'gameover_music.mp3'))
    pygame.mixer.music.play()
    pygame.mixer.music.set_volume(0.01)

    # draw game over
    gameOverFont = pygame.font.Font('freesansbold.ttf', 100)
    gameSurf = gameOverFont.render('Game', True, WHITE)
    overSurf = gameOverFont.render('Over', True, WHITE)
    gameRect = gameSurf.get_rect()
    overRect = overSurf.get_rect()
    gameRect.midtop = (WIDTH / 2, 10)
    overRect.midtop = (WIDTH / 2, gameRect.height + 10 + 25)

    screen.blit(gameSurf, gameRect)
    screen.blit(overSurf, overRect)

    drawPressKeyMsg()

    pygame.display.update()
    pygame.time.wait(1000)

    while True:
        if checkForKeyPress():
            return


# game win
def game_win():

    # draw background
    screen.blit(bg_img, (0, 0))

    myFont = pygame.font.Font(None, 80)
    loadImage = myFont.render("YOU WIN. Press SPACE to Start", True, WHITE)
    screen.blit(loadImage, (550, 1000))

    player.draw(screen)

    pygame.display.update()
    pygame.time.wait(1000)

    while True:
        if checkForKeyPress():
            return


# process pause
def pause():

    start_x = 700
    start_y = 500
    gap = 100
    length = 600

    myFont = pygame.font.Font(None, 80)

    pauseImage = myFont.render("Back to main Menu", True, WHITE)
    screen.blit(pauseImage, (start_x, start_y))

    pauseImage = myFont.render("Back to Game", True, WHITE)
    screen.blit(pauseImage, (start_x, start_y + gap))

    pygame.display.update()

    while True:
        check_quit()

        loc = check_click()
        if loc is not None:
            if loc[0] in range(start_x, start_x + length) and loc[1] in range(500, 590):  # Back to main Menu
                save()
                main()
                terminate()

            if loc[0] in range(start_x, start_x + length) and loc[1] in range(590, 680):  # Back to Game
                return


# close the game
def terminate():
    #save()
    pygame.quit()
    sys.exit()


def run():

    global screen, player, enemies, clock, count, start, switch, score

    # clear up
    props.empty()
    enemies.empty()
    all_boss.empty()
    pygame.event.clear()

    # reset score counter
    score = 0

    while True:

        clock.tick(FPS)

        # check switch music
        if switch:
            pygame.mixer.music.load(os.path.join(music_folder, 'bg_music.mp3'))
            pygame.mixer.music.play()
            pygame.mixer.music.set_volume(0.03)
            switch = False

        # Process input (events)
        check_quit()

        # check win
        if check_win():
            game_win()
            save()
            main()
            terminate()

        # check jump
        if check_jump():
            player.jump()

        # check shot
        if check_shot() and count % 15 == 0 and player.load > 0:
            bullet = Bullet(player)
            player.bullets.add(bullet)
            player.load -= 1

        # check pause
        if check_esc():
            pause()

        # player moving
        player.move(check_move())

        # randomly create enemies and props
        create_enemy()
        create_prop()

        # check collision
        check_collision()

        # check game over
        if player.hp < 0:
            game_over()
            save()
            main()
            terminate()

        # Update
        player.update()
        enemies.update()
        all_boss.update()
        props.update()

        #  get message that need to present

        fpsImage = drawFPS()
        scoreImage = drawScore()

        # Draw / Render
        screen.blit(bg_img, (0, 0))

        player.draw(screen)
        for enemy in enemies:
            enemy.draw(screen)
        for boss in all_boss:
            boss.draw(screen)
        for prop in props:
            prop.draw(screen)

        getAllBullets().draw(screen)
        drawHUD(screen)
        # screen.blit(fpsImage, (0, 0))
        screen.blit(scoreImage, (1700, 0))

        pygame.display.flip()


def drawFPS():
    global count
    count += 1
    now = time.time()
    elapse = now - start
    fpsfont = pygame.font.Font(None, 25)
    fps = count / elapse if elapse != 0 else 0

    fpsImage = fpsfont.render("FPS: " + str(fps), True, WHITE)
    return fpsImage


def drawScore():
    scoreFont = pygame.font.Font(None, 50)
    scoreImage = scoreFont.render("Score: " + str(score), True, WHITE)
    return scoreImage


def drawHUD(surface):
    surface.blit(player.image, (50, 1050))

    def set_color(perc):
        if perc < 0.5:
            res = RED
        elif perc > 0.9:
            res = GREEN
        else:
            res = YELLOW
        return res

    bg_color = BLACK

    # Draw Hp information
    # surface.blit(hpBar_img, (192, 1037))

    hpFont = pygame.font.Font(None, 35)
    hpImage = hpFont.render("   Hp: ", True, WHITE)
    screen.blit(hpImage, (155, 1070))

    percent = player.hp / player.hp_full * 1.0
    pygame.draw.rect(surface, bg_color, (250, 1080, 15 * player.hp_full, 8))
    color = set_color(percent)
    pygame.draw.rect(surface, color,
                     (250, 1080, 15 * player.hp * percent + 10, 8))

    # Draw Amo information
    amoFont = pygame.font.Font(None, 35)
    amoImage = amoFont.render("Amo: ", True, WHITE)
    screen.blit(amoImage, (155, 1100))

    percent = player.load / player.load_full * 1.0
    pygame.draw.rect(surface, bg_color, (250, 1110, 0.3 * player.load_full, 10))
    color = set_color(percent)
    pygame.draw.rect(surface, color,
                     (250, 1110, 0.3 * player.load * percent, 10))

    # Draw fuel information
    fuelFont = pygame.font.Font(None, 35)
    fuelImage = amoFont.render("Fuel: ", True, WHITE)
    screen.blit(fuelImage, (155, 1130))

    percent = player.fuel / player.fuel_full * 1.0
    pygame.draw.rect(surface, bg_color, (250, 1140, 0.02 * (player.fuel_full - 2), 10))
    color = set_color(percent)
    pygame.draw.rect(surface, color,
                     (250, 1140, 0.02 * player.fuel * percent, 10))


def drawPressKeyMsg():
    myfont = pygame.font.Font(None, 80)
    loadImage = myfont.render("Press SPACE to Continue", True, WHITE)
    screen.blit(loadImage, (650, 1000))


def load(filename="data.txt"):

    with open(filename, "r") as file:
        data = file.readlines()
    for line in data:
        line = int(line.strip('\n'))

    player.hp_full = int(data[0])
    player.hp = int(data[0])
    player.load_full = int(data[1])
    player.fuel_full = int(data[2])


def get_score(filename="data.txt"):

    with open(filename, "r") as file:
        data = file.readlines()
    for line in data:
        line = int(line.strip('\n'))

    return data[4]


def save(filename="data.txt"):
    data = None
    with open(filename, "r") as file:
        data = file.readlines()
        for line in data:
            line = int(line.strip('\n'))
    file.close()

    with open(filename, "w") as file:
        file.write(str(player.hp_full) + '\n')
        file.write(str(player.load_full) + '\n')
        file.write(str(player.fuel_full) + '\n')
        file.write(str(player.speed_limit) + '\n')
        file.write(str(int(data[4]) + score) + '\n')


if __name__ == '__main__':
    main()
